pysma/__init__.py

```diff
@@ -426,11 +431,22 @@ async def device_info(self):
             "serial": self.device_info_sensors["serial_number"].value
             or fallback_device_info["serial"],
             "name": self.device_info_sensors["device_name"].value
-            or fallback_device_info["senamerial"],
+            or fallback_device_info["name"],
             "type": self.device_info_sensors["device_type"].value
             or fallback_device_info["type"],
             "manufacturer": self.device_info_sensors["device_manufacturer"].value
             or fallback_device_info["manufacturer"],
         }
 
         return device_info
+
+    async def get_devclass(self, result_body):
+        if not self.devclass:
+            sensor_value = list(result_body.values())[0]
```

Comment:
What if the result_body was empty? Then you cannot access [0]?

Reply:
You are right, I have updated it to return None in that case.